### PR TITLE
feat(advanced-metrics): Add Support for Topology (Availability) Zone Labels

### PR DESCRIPTION
**Pull request by rickardsjp**
# Description

This PR introduces availability zone labels ([topology.kubernetes.io/zone](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone)) for remote context advanced metrics.

With this PR, it is possible to add the `zone` to metrics as sourceLabel or destinationLabel via the MetricsConfiguration CRD.

The data is added via the enricher component to the RetinaMetadata in the Flow struct's Extensions attribute, because the node controller already has access to the labels of the Nodes.

## Related Issue

https://github.com/microsoft/retina/issues/1654

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/Contributing/overview).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

img

## Additional Notes

We have not added it to the local context because we don't see the value in adding just one zone to a pod. We have not added it to the hubble scenario because hubble doesn't support availability zones for the destination yet.
Tagging the co-authors: @ScheererJ @rrhubenov

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

**Comment by rickardsjp** [comment]
@microsoft-github-policy-service agree [company="SAP SE"]

**Review by SRodi — COMMENTED** [review]
Thanks for this PR @rickardsjp! Can you please make sure all the commits are [cryptographically signed](https://retina.sh/docs/Contributing/development#cryptographic-signing-of-commits)?

**Comment by rickardsjp** [comment]
Done! Thanks for re-running the CI for all the force pushes, we had issues running some of the makefile targets locally. I also can't figure out why the Windows image builds fail - any pointers? I can't really make sense of the `503` and `EOF`s.

**Comment by nddq** [comment]
@rickardsjp MCR is acting up for windows 2019 images, but it should be transient

**Review by SRodi — COMMENTED** [review]
Thanks @rickardsjp ! Just a few general comments:

1. can we have test coverage for negative/edge cases, like missing or malformed zones?
2. is there any backward compatibility consideration and potential impact on downstream consumers for RetinaEndpoint? (I do not think so, but have you considered/tested that?)
3. shall we make the default zone "unknown" a configurable value? and maybe add logging for missing zones?

File: pkg/enricher/enricher.go [hunk 1]
```diff
@@ -148,6 +149,19 @@ func (e *Enricher) enrich(ev *v1.Event) {
...
+	dstZone := e.zoneFromObj(dstObj)
+
+	if ext := utils.GetExtensionsStruct(flow); ext != nil {
+		utils.AddZones(ext, srcZone, dstZone)
+		utils.SetExtensions(flow, ext)
+	} else {
+		ext := utils.NewExtensions()
+		utils.AddZones(ext, srcZone, dstZone)
+		utils.SetExtensions(flow, ext)
+	}
```
Comment (nddq) [inline pkg/enricher/enricher.go]:
```suggestion
	ext := utils.GetExtensionsStruct(flow)
    if ext == nil {                                                                                                               
       ext = utils.NewExtensions()                                                                                               
    }                              
    utils.AddZones(ext, srcZone, dstZone)                                                                                         
    utils.SetExtensions(flow, ext)
```

Comment (rickardsjp) [reply pkg/enricher/enricher.go]: good point, fixed.
